### PR TITLE
Fix ~PcapFileWriterDevice() to close the pcap file.

### PR DESCRIPTION
**Pull request by rndx21033**
Empty destructor could leave file partially written.  Fix to close on destruction.

### CODE REVIEW

**Review by clementperon — APPROVED** [review]

**Review by Dimi1010 — APPROVED** [review]

File: Tests/Pcap++Test/Tests/FileTests.cpp [hunk 1]
```diff
@@ -974,3 +974,44 @@ PTF_TEST_CASE(TestSolarisSnoopFileRead)
...
+	// Check that file sizes are equal.  This may fail if the pcpp::PcapFileWriterDevice destructor does not close
+	FILE* writerDevDestructor1 = fopen(EXAMPLE_PCAP_DESTRUCTOR1_PATH, "rb");
+	fseek(writerDevDestructor1, 0, SEEK_END);
+	long pos1 = ftell(writerDevDestructor1);
+	fclose(writerDevDestructor1);
+	FILE* writerDevDestructor2 = fopen(EXAMPLE_PCAP_DESTRUCTOR2_PATH, "rb");
+	fseek(writerDevDestructor2, 0, SEEK_END);
+	long pos2 = ftell(writerDevDestructor2);
+	fclose(writerDevDestructor2);
+	PTF_ASSERT_EQUAL(pos1, pos2);
```
Comment (seladb) [inline Tests/Pcap++Test/Tests/FileTests.cpp:1016]: Why not use C++11 `ifstream`, `seekg` and `tellg`?

File: Tests/Pcap++Test/Tests/FileTests.cpp [hunk 2]
```diff
@@ -974,3 +974,44 @@ PTF_TEST_CASE(TestSolarisSnoopFileRead)
 
 	readerDev.close();
 }  // TestSolarisSnoopFileRead
+
+PTF_TEST_CASE(TestPcapFileWriterDeviceDestructor)
```
Comment (tigercosmos) [inline Tests/Pcap++Test/Tests/FileTests.cpp:978]: It's a bit hard to read and understand your intention. Please simplify the test, e.g. reduce the writes, and add more comments.

File: Tests/Pcap++Test/Tests/FileTests.cpp [hunk 3]
```diff
@@ -974,3 +974,44 @@ PTF_TEST_CASE(TestSolarisSnoopFileRead)
...
+{
+	std::array<uint8_t, 16> testPayload = { 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+		                                    0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F };
+	pcpp::RawPacket rawPacket1(testPayload.data(), testPayload.size(), timeval({ 1, 2 }), false);      // 1.000002000
+	pcpp::RawPacket rawPacket2(testPayload.data(), testPayload.size(), timespec({ 1, 1234 }), false);  // 1.000001234
+
+	// Create some pcaps in a nested scope to test cleanup on destruction.
+	{
+		pcpp::PcapFileWriterDevice writerDevDestructor1(EXAMPLE_PCAP_DESTRUCTOR1_PATH, pcpp::LINKTYPE_ETHERNET, false);
+		PTF_ASSERT_EQUAL(writerDevDestructor1.getTimestampPrecision(), pcpp::FileTimestampPrecision::Microseconds,
```
Comment (tigercosmos) [inline Tests/Pcap++Test/Tests/FileTests.cpp:988]: Do we need this line?

**Comment by seladb** [comment]
@rndx21033 will you have time to address the PR comments?

**Comment by rndx21033** [comment]
> @rndx21033 will you have time to address the PR comments?

I should have time to address them this weekend.

File: Tests/Pcap++Test/Tests/FileTests.cpp (hunk 1 above)
Comment (rndx21033) [reply Tests/Pcap++Test/Tests/FileTests.cpp:1016]: Updated to use those.

**Review by seladb — APPROVED** [review]

**Pull request merged**